library/ssl_msg.c

```diff
@@ -377,28 +377,72 @@ static int ssl_parse_inner_plaintext( unsigned char const *content,
 }
 #endif /* MBEDTLS_SSL_DTLS_CONNECTION_ID || MBEDTLS_SSL_PROTO_TLS1_3 */
 
-/* `add_data` must have size 13 Bytes if the CID extension is disabled,
- * and 13 + 1 + CID-length Bytes if the CID extension is enabled. */
+/* The size of the `add_data` structure depends on various
```

Comment:
Are those changes enough? The upstream PR introduces much more changes, including some incompatible with https://github.com/zephyrproject-rtos/mbedtls/pull/36/commits/2c79e20b4bc3cc669abf422caad424bcd8a2dfd3

Reply:
The original author made a number of changes a few weeks ago, so I need to revisit this.

Reply:
I just replaced the commit from Mbed-TLS upstream with the one from the new PR from Hannes that replaces the old one.